pkg/sql/catalog/tabledesc/column.go

```diff
@@ -187,6 +187,12 @@ func (w column) IsSystemColumn() bool {
 	return w.desc.SystemColumnKind != descpb.SystemColumnKind_NONE
 }
 
+// IsGeneratedAsIdentity returns true iff the column is created
+// with GENERATED {ALWAYS | BY DEFAULT} AS IDENTITY token.
+func (w column) IsGeneratedAsIdentity() bool {
+	return w.desc.GeneratedAsIdType != descpb.GeneratedAsIdentityType_NOT_IDENTITY_COLUMN
+}
+
```

Comment:
Perhaps add another method that returns `w.desc.GeneratedAsIdType`? This way you don't need to retrieve the enum value via `.ColumnDesc()` when building the opt catalog. which is something we want to discourage.

Reply:
In fact looking at your subsequent PRs I noticed that you added these in a subsequent commit. Perhaps move them into this commit? Your subsequent changes in `pkg/sql/catalog/schemaexpr/column.go` would also belong in this commit.